src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNICommon.Windows.cs

```diff
@@ -0,0 +1,12 @@
+using System;
+
+namespace System.Data.SqlClient.SNI
+{
+    internal partial class SNICommon
+    {
+        public static string GetNativeErrorMessage(int error)
+        {
+            return Interop.Kernel32.GetMessage(error);
+        }
+    }
+}
```

Comment:
You shouldn't need these P/Invokes... you can just do:
```C#
string message = new Win32Exception(error).Message;
```
Presumably this only happens on error paths where the extra allocation for the exception object won't matter.  Win32Exception itself calls either StrError or GetMessage:
https://github.com/dotnet/corefx/blob/master/src/Microsoft.Win32.Primitives/src/System/ComponentModel/Win32Exception.Unix.cs#L9
https://github.com/dotnet/corefx/blob/master/src/Microsoft.Win32.Primitives/src/System/ComponentModel/Win32Exception.Windows.cs#L9
Win32Exception's ctor also calls Marshal.GetLastWin32Error, so you could skip calling that manually below, too.